devicemapper/thin_pool_watcher.go

```diff
@@ -163,14 +163,14 @@ func (w *ThinPoolWatcher) checkReservation(poolName string) (bool, error) {
 		return false, err
 	}
 
-	tokens := strings.Split(string(output), " ")
-	// Split returns the input as the last item in the result, adjust the
-	// number of tokens by one
-	if len(tokens) != thinPoolDmsetupStatusTokens+1 {
-		return false, fmt.Errorf("unexpected output of dmsetup status command; expected 11 fields, got %v; output: %v", len(tokens), string(output))
+	// we care about the field at fields[thinPoolDmsetupStatusHeldMetadataRoot],
+	// so make sure we get enough fields
+	fields := strings.SplitN(string(output), " ", thinPoolDmsetupStatusMinTokens+1)
```

Comment:
I'll take a look at Fields
On Thu, Jun 23, 2016 at 2:25 PM Tim St. Clair <notifications@github.com>
wrote:

> In devicemapper/thin_pool_watcher.go
> <https://github.com/google/cadvisor/pull/1348#discussion_r68288542>:
>
> > @@ -163,14 +163,14 @@ func (w *ThinPoolWatcher) checkReservation(poolName string) (bool, error) {
> >  		return false, err
> >  	}
> >
> > -	tokens := strings.Split(string(output), " ")
> > -	// Split returns the input as the last item in the result, adjust the
> > -	// number of tokens by one
> > -	if len(tokens) != thinPoolDmsetupStatusTokens+1 {
> > -		return false, fmt.Errorf("unexpected output of dmsetup status command; expected 11 fields, got %v; output: %v", len(tokens), string(output))
> > +	// we care about the field at fields[thinPoolDmsetupStatusHeldMetadataRoot],
> > +	// so make sure we get enough fields
> > +	fields := strings.SplitN(string(output), " ", thinPoolDmsetupStatusMinTokens+1)
>
> Would the fields ever be split by tabs or multiple spaces? If so
> strings.Fields would be a better choice.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/cadvisor/pull/1348/files/8faba9efbd4dcd245225157a10cf2b8bcf565c56#r68288542>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAABYsrTqz03aQYzk1KQqoh8oRxip3jRks5qOs-bgaJpZM4I87rG>
> .
>